### PR TITLE
funcs to fetch details from cluster provision status

### PR DESCRIPTION
**Pull request by vprabhakar-px**
changes to provision staus functions to pick details on the pool

### CODE REVIEW

File: tests/common.go [hunk 1]
```diff
@@ -9198,6 +9191,38 @@ func GetClusterProvisionStatus() ([]ProvisionStatus, error) {
...
+	// Using Node which is up and running
+	var selectedNode []node.Node
+	for _, eachNode := range node.GetStorageDriverNodes() {
+		status, err := IsPxRunningOnNode(&eachNode)
+		if err != nil {
+			log.InfoD("Px is not running on the Node.. searching for other node")
+			continue
+		}
+		if status {
+			selectedNode = append(selectedNode, eachNode)
```
Comment (lsrinivas-pure) [inline tests/common.go:9209]: can we select one node and break this loop? any particular reason you want all nodes and then select random node?

File: tests/common.go [hunk 2]
```diff
@@ -9295,6 +9322,39 @@ func WaitForSnapShotToReady(snapshotScheduleName, snapshotName, appNamespace str
 
 }
 
+// WaitTillPoolExpanded waits till pool expansion gets completed
+func WaitTillPoolExpanded(poolUUID string, poolCurrSize uint64) error {
```
Comment (lsrinivas-pure) [inline tests/common.go:9326]: How are you making pool expansion is completed here? Sometimes pool size changes but rebalance might be in-progress

File: tests/common.go (hunk 1 above)
Comment (vprabhakar-px) [reply tests/common.go:9209]: oh !! I thought I was breaking it , let me reupdate it similar to above ones

Comment (vprabhakar-px) [reply tests/common.go:9209]: the intension here is to pick up random node , for specific node we already have a function GetClusterProvisionStatusOnSpecificNode which can be used

File: tests/common.go (hunk 2 above)
Comment (vprabhakar-px) [reply tests/common.go:9326]: yes, you are right , I missed that part , will update it . currently I am checking only till pool size got increased

Comment (vprabhakar-px) [reply tests/common.go:9326]: removing this function as we already have waitForPoolToBeResized

File: tests/common.go [hunk 3]
```diff
@@ -9200,6 +9193,39 @@ func GetClusterProvisionStatus() ([]ProvisionStatus, error) {
...
+	// Select Random Volumes for pool Expand
+	randomIndex := rand.Intn(len(selectedNode))
+	randomNode := selectedNode[randomIndex]
+
+	clusterProvision, err := GetClusterProvisionStatusOnSpecificNode(randomNode)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("Cluster provision status [%v]", clusterProvision)
```
Comment (lsrinivas-pure) [inline tests/common.go:9228]: this is getting logged twice once here and once in  GetClusterProvisionStatusOnSpecificNode

**Review by lsrinivas-pure — APPROVED** [review]
LGTM , one minor comment

**Pull request merged**